Android/app/src/main/java/app/intra/net/go/GoVpnAdapter.java

```diff
@@ -106,10 +109,22 @@ public synchronized void start() {
     listener = new GoIntraListener(vpnService);
     String dohURL = PersistentState.getServerUrl(vpnService);
 
+    Transport transport;
+    if (RemoteConfig.getUseGoDoh()) {
+      try {
+        transport = makeDohTransport(dohURL);
+      } catch (Exception e) {
+        // Fall back to ServerConnection if Go-DOH setup fails.
+        // TODO: Report this failure properly instead of silently falling back.
+        LogWrapper.logException(e);
+        transport = null;
+      }
+    } else {
```

Comment:
`transport` is already `null` if `!RemoteConfig.useGoDoh()`.

Reply:
OK, I made this assignment more compact.